File: eShop/eShop/Models/Item.cs
namespace eShop.Models
{
    public class Item
    {
        public Guid Id { get; set; }
        public required string Title { get; set; }
        public required string Description { get; set; }
        public decimal Price { get; set; }
        public required DiscountPolicy DiscountPolicy { get; set; }
        public decimal Rating { get; set; }
        public int Stock { get; set; }
        public required Brand Brand { get; set; }
        public required Category Category { get; set; }
        public required string Thumbnail { get; set; }
        public List<string>? Images { get; set; }
    }
}

File: eshop_react_frontend/src/Entities/Item.ts
export interface IItem{
    title : string;
    description : string;
    price : number;
    discountPolicy : IDiscountPolicy;
    rating : number;
    stock : number;
    brand : IBrand;
    category : ICategory;
    thumbnail : string;
    images : Array<string>;
}

export interface IDiscountPolicy{
    discountPolicy : number;
}

export interface IBrand{
    displayName : string;
}

export interface ICategory{
    displayName : string;
}